### ISSUE TITLE
client will receive duplicate messages when overlapping subscription

### ISSUE THREAD

**Issue by erylee** [emqtt/emqttd#53]
{ok, C} = emqttc:start_link().
emqttc:subscribe(C, <<"t">>, 1).
emqttc:subscribe(C, <<"t">>, 2).

{ok, C2} = emqttc:start_link().
emqttc:publish(C2, <<"t">>, <<"hello">>, [{qos, 2}]).

**Comment by kansifang** [comment]
我也有这个问题，当同一个TOPIC订阅两次就会收到两条信息

**Comment by kansifang** [comment]
I have the similar problem that is a client subscribe a topic twice or three times,when publish a message to the topic, the client will receive  twice or three messages。

**Comment by kansifang** [comment]
This is a bug or can throuth config to void this problem?

**Comment by erylee** [comment]
@kansifang, do you subscribe the same topic with same Qos? 

For example, if you subscribe with same Qos:
```
{ok, C} = emqttc:start_link().
emqttc:subscribe(C, <<"t">>, 1).
emqttc:subscribe(C, <<"t">>, 1).
```

client should receive only one message.

**Comment by sailingchen** [comment]
subscribe the topic , client:
1.  disable 'clean session'
2. qos set  1
3. no retained message
4.  set the client_id

example for mosquitto:
mosquitto_sub   -c -i 0000aaeef -q 1   -t a/b/c

note: don't use the -r option.
tested  it .

**Comment by erylee** [comment]
@huangdan please check the issue

**Comment by huangdan** [comment]
@kansifang 不好意思回复晚了点，mosquitto_sub -c -i 0000aaeef -q 1 -t a/b/c 这个订阅有什么问题吗？

**Comment by erylee** [comment]
MQTT V3.1.1 [MQTT-3.8.4-3]:

```
If a Server receives a SUBSCRIBE Packet containing a Topic Filter that is identical to an existing Subscription’s Topic Filter then it MUST completely replace that existing Subscription with a new Subscription. The Topic Filter in the new Subscription will be identical to that in the previous Subscription, although its maximum QoS value could be different. Any existing retained messages matching the Topic Filter MUST be re-sent, but the flow of publications MUST NOT be interrupted [MQTT-3.8.4-3].
```

**Comment by erylee** [comment]
@kansifang, @sailingchen, this issue is fixed in 0.8.5. The broker will remove the overlapping subscriptions with different QoS.

**Comment by saqib-cielo** [comment]
Emqx as a bridge with AWS Iot broker behaving unexpectedly(getting messages burst with any Qos level), normally happen when we start publishing messages with some pace, check all config params but no luck, did anyone came across similar issue.
